src/ImageSharp/Processing/Quantization/IQuantizer.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing.Dithering.ErrorDiffusion;
+using SixLabors.ImageSharp.Processing.Quantization.FrameQuantizers;
+
+namespace SixLabors.ImageSharp.Processing.Quantization
+{
+    /// <summary>
+    /// Provides methods for allowing quantization of images pixels with configurable dithering.
+    /// </summary>
+    public interface IQuantizer
+    {
+        /// <summary>
+        /// Gets a value indicating whether to apply dithering to the output image.
+        /// </summary>
+        bool Dither { get; }
+
+        /// <summary>
+        /// Gets the dithering algorithm to apply to the output image.
+        /// </summary>
+        IErrorDiffuser DitherType { get; }
```

Comment:
not really needed on the interface as its an implementation details inside the `XXXFrameQuantizer`
noting can set it so it will only be configured during construction of a concrete type.

Reply:
We'll need this to pass to the `FrameQuantizerBase{TPixel}` class as the constructor takes the `IQuantizer` interface. You have got me thinking though about the `Dither` property. We don't need it as long as this is set but we'll have to reverse the default behavior in order to avoid the dev having to pass `null` to a constructor (which I find icky). Currently the default constructor diffuses.

e.g 

- `new OctreeQuantizer()` **255 colors, no diffusion**
- `new OctreeQuantizer(128)` **28 colors, no diffusion**
- `new OctreeQuantizer(DiffuseMode.FloydSteinberg)` **255 colors, FloydSteinberg.Diffusion**
- `new OctreeQuantizer(DiffuseMode.FloydSteinberg, 128)` **128 colors, FloydSteinberg.Diffusion**

I'll probably change the `DitherType` name to `DiffuseMode` also to match the dithering API.